plugins/passwordagentauthplugin/passwordagentplugin.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2017 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "passwordagentplugin.h"
+
+#include <QDBusAbstractAdaptor>
+#include <QDBusConnection>
+#include <QDBusMessage>
+#include <QDBusPendingCallWatcher>
+#include <QLoggingCategory>
+
+Q_PLUGIN_METADATA(IID Sailfish_Secrets_AuthenticationPlugin_IID)
+
+namespace Sailfish {
+
+namespace Secrets {
+
+namespace Daemon {
+
+namespace Plugins {
+
+Q_DECLARE_LOGGING_CATEGORY(lcPasswordAgent)
+Q_LOGGING_CATEGORY(lcPasswordAgent, "org.sailfishos.secrets.plugin.authentication.passwordagent", QtWarningMsg)
+
+static QDBusMessage passwordAgentMethodCall(const QString &member)
+{
+    return QDBusMessage::createMethodCall(
+                QStringLiteral("org.sailfishos.Lipstick.SecurityUi"),
+                QStringLiteral("/org/sailfishos/Lipstick/SecurityUi/PasswordAgent"),
+                QStringLiteral("org.sailfishos.Security.PasswordAgent"),
+                member);
+}
+
+class PasswordAgentResponse : public QDBusPendingCallWatcher
+{
+public:
+    PasswordAgentResponse(
+            const QDBusObjectPath &object,
+            const QDBusPendingCall &call,
+            uint callerPid,
+            qint64 requestId,
+            const QString &callerApplicationId,
+            const QString &collectionName,
+            const QString &secretName,
+            const QString &interactionServiceAddress,
+            QObject *parent)
+        : QDBusPendingCallWatcher(call, parent)
+        , callerApplicationId(callerApplicationId)
+        , collectionName(collectionName)
+        , secretName(secretName)
+        , serviceAddress(interactionServiceAddress)
+        , requestId(requestId)
+        , callerPid(callerPid)
+    {
+        QDBusConnection::systemBus().registerObject(
+                    object.path(),
+                    QStringLiteral("org.sailfishos.Security.PasswordClient"),
+                    this,
+                    QDBusConnection::ExportAllSlots);
+    }
+
+    void cancel()
+    {
+        QDBusConnection::systemBus().send(
+                    passwordAgentMethodCall(QStringLiteral("Cancel")) << QVariant::fromValue(object));
+    }
+
+public slots:
+    void VerifyPassword(const QString &password)
```

Comment:
It's a an implementation of a DBus interface and the convention for dbus apis is to name members using an upper camel case.

It's called VerifyPassword because the caller would prefer that some validation was done on the password, and a dbus error be returned if it is incorrect so it can prompt for the user to retry entering the password. There's no ability to verify the password here so we accept it irrespective and let the dbus function return to the caller.

Reply:
Okay